### ISSUE TITLE
Compiler sub-package npm dependency works locally, but not from atmospherejs

### ISSUE THREAD

**Issue by kamilkisiela** [meteor/meteor#7069]
### TL;DR

`html-minifier` as one of `Npm.depends` of package X won't work inside Plugin of package Y.
### Description

I have two packages:
- one with class that depends on `html-minifier`
- second package registers a plugin that depends on first package and uses its class.

It works locally, but not when it comes from the atmospherejs.

`html-minifier` module works fine when it's a direct dependency of the second package or if it's being used inside meteor app directly. So probably there is nothing wrong with the node module.

You can see the actual error inside repository with a reproduction.
### Reproduction

[Repo with repro](https://github.com/kamilkisiela/meteor-html-minifier)
### Environment

Related to `Meteor@1.3+` and `atmospherejs`.
### Example

Package with compiler:

``` js
Npm.depends({
  'html-minifier': '2.1.3' // works if downgraded to 0.X version
});

Package.onUse(function(api) {
  api.versionsFrom('1.3');
  api.use('ecmascript@0.2.0');
  api.use('caching-compiler@1.0.2');

  api.mainModule('compiler.js', 'server');

  api.export([
    'ExampleCompiler'
  ], 'server');
});
```

Plugin:

``` js
Package.registerBuildPlugin({
  name: 'htmlMinifer',
  use: [
    'ecmascript@0.2.0',
    'package-with-compiler@0.0.2' // it's here
  ],
  sources: [
    'plugin.js'
  ]
});

Package.onUse(function(api) {
  api.versionsFrom('1.3');
  api.use('isobuild:compiler-plugin@1.0.0');
});
```

**Comment by laosb** [comment]
Someone should take a look or I will do it this weekend.

**Comment by abernix** [comment]
I took a super quick look but nothing jumped out at me right away.  Does it work fine if you just set it up as a single package instead of multiple? (and just use the `npmDependencies` option to `Package.registerBuildPlugin`?

**Comment by kamilkisiela** [comment]
@abernix You can just use `mys:hm` package and it should throw an error.

I [created](https://github.com/kamilkisiela/meteor-html-minifier/commit/be70905f1e7f3aa92e83f500cb632c54cb416d8e) a package that is a combination of both I mentioned, so I put everything inside `Plugin`. It works. 

But that's not the case. It's not even a workaround because I want to expose only that class in a package, not with the plugin.

**Comment by abernix** [comment]
Ok, I can confirm that based on your repo it does happen when using the deployed atmosphere package but NOT when dropping the packages in locally to `packages/`.  Does that sound correct?

**Comment by kamilkisiela** [comment]
Yes, exactly :)

**Comment by kamilkisiela** [comment]
Quick update. It happens locally too but after few app reloads so maybe it's related to some caching thing.

**Comment by kamilkisiela** [comment]
@benjamn It seems like it's [working now](https://github.com/kamilkisiela/meteor-html-minifier/commit/64be05c78a01938c9af1ecad84eeb9bec2fee731), but could you confirm that?

If so, we can close this issue.

**Comment by hwillson** [comment]
Closing as this issue appears to be fixed with recent versions of Meteor. Thanks for reporting this originally!
